Yosemite/Yosemite/PointOfSale/PointOfSaleCouponService.swift

```diff
@@ -0,0 +1,81 @@
+import protocol Networking.Network
+import protocol Networking.ProductVariationsRemoteProtocol
+import class Networking.ProductsRemote
+import class Networking.ProductVariationsRemote
+import class Networking.AlamofireNetwork
+import class WooFoundation.CurrencyFormatter
+import class WooFoundation.CurrencySettings
+import Storage
+
+public final class PointOfSaleCouponService: PointOfSaleItemServiceProtocol {
+    private var siteID: Int64
+    private let currencyFormatter: CurrencyFormatter
+    private let productsRemote: ProductsRemote
+    private let variationRemote: ProductVariationsRemoteProtocol
+    private let storage: StorageManagerType?
+
+    public init(siteID: Int64,
+                currencySettings: CurrencySettings,
+                network: Network,
+                storage: StorageManagerType? = nil) {
+        self.siteID = siteID
+        self.currencyFormatter = CurrencyFormatter(currencySettings: currencySettings)
+        self.productsRemote = ProductsRemote(network: network)
+        self.variationRemote = ProductVariationsRemote(network: network)
```

Comment:
I've kept this as it is for now, but we will need to split the `PointOfSaleItemServiceProtocol`, as coupons do not need to know about some of these dependencies.

Reply:
No need for splitting.

A separate `PointOfSaleCouponServiceProtocol ` is enough. We don't need the ability to interchangeably use `PointOfSaleCouponService` and `PointOfSaleItemService`. So we can safely remove all those methods and have:

```
public protocol PointOfSaleCouponServiceProtocol {
    func providePointOfSaleCoupons(pageNumber: Int) async throws -> PagedItems<POSItem>
}
```

Reply:
Gotcha! I'll take a look into this in the next PR as I start to implement those.